spec/compiler/type_inference/generic_class_spec.cr

```diff
@@ -243,19 +243,6 @@ describe "Type inference: generic class" do
       )) { types["MyType"].types["Bar"] }
   end
 
-  it "error on T::Type with T a generic type that's a union" do
-    assert_error %(
-      class Foo(T)
-        def self.bar
-          T::Bar
-        end
-      end
-
-      Foo(Char | String).bar
-      ),
-      "can't lookup type in union (Char | String)"
-  end
```

Comment:
If you do `UnionType::T` it gives you the `T` (the union type). Otherwise it searches in the program. Originally it gave an error when not using `T`, but the problem is that inside a `Union` method one wants `Int32` and other types to be found.

Hmmm... it's a border case which I think doesn't matter, but now that you mention it, I think it can be done in a better way. I'll try to revive that spec. In fact, the new error was "undefined constant T::Bar" or something like that, which is the correct error message after all. Thanks for checking this, @jhass